blockchain/types/transaction_signing.go

```diff
@@ -306,16 +438,17 @@ func (s EIP155Signer) SenderFeePayer(tx *Transaction) ([]*ecdsa.PublicKey, error
 	})
 }
 
-// WithSignature returns a new transaction with the given signature. This signature
+// SignatureValues returns a new transaction with the given signature. This signature
 // needs to be in the [R || S || V] format where V is 0 or 1.
-func (s EIP155Signer) SignatureValues(sig []byte) (R, S, V *big.Int, err error) {
-	if len(sig) != crypto.SignatureLength {
-		panic(fmt.Sprintf("wrong size for signature: got %d, want %d", len(sig), crypto.SignatureLength))
+func (s EIP155Signer) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big.Int, err error) {
+	if tx.Type().IsEthTypedTransaction() {
+		return nil, nil, nil, ErrTxTypeNotSupported
 	}
-	R = new(big.Int).SetBytes(sig[:32])
-	S = new(big.Int).SetBytes(sig[32:64])
-	V = big.NewInt(int64(sig[crypto.RecoveryIDOffset] + 35))
+
+	R, S, _ = decodeSignature(sig)
+	V = big.NewInt(int64(sig[64] + 35))
```

Comment:
Can I know why you don't use `RecoveryIDOffset` variable? 
```suggestion
	V = big.NewInt(int64(sig[crypto.RecoveryIDOffset] + 35))
```